pippy/PipelineStage.py

```diff
@@ -653,6 +671,93 @@ def forward_one_chunk(
 
         return output, send_reqs
 
+    def forward_one_chunk_ipipe(
+        self,
+        chunk: int,
+        args_split,
+        kwargs_split,
+        fwd_cache: Dict[int, Any],
+    ):
+        if self.rank == self.nstages - 1:
```

Comment:
If I understand correctly, `self.nstages` is the global stages?

Reply:
```self.nstages``` has been used in original codebase so i added a new variable ```self.global_depth``` to store global stages. so ```self.nstages``` is same as pp_group_size. (# of ranks)